### PR TITLE
allow usage of underscore in external libraries rules

### PR DESCRIPTION
**Pull request by irengrig**

### CODE REVIEW

File: tools/build_defs/framework.bzl [hunk 1]
```diff
@@ -249,8 +249,9 @@ def _check_file_name(var, name):
 
   if (not var[0:1].isalpha()):
     fail("{} should start with a letter.".format(name.capitalize()))
-  if (not var.isalnum()):
-    fail("{} should be alphanumeric.".format(name.capitalize()))
+  for letter in var:
```
Comment (dslomov) [inline tools/build_defs/framework.bzl]: iterable strings are going away in Skylark, please use a for loop over indices

Comment (irengrig) [reply tools/build_defs/framework.bzl]: Corrected

**Comment by dslomov** [comment]
why so many changes? did you reformat with buildifier?

File: tools/build_defs/framework.bzl (hunk 1 above)
Comment (irengrig) [reply tools/build_defs/framework.bzl]: Yep, actually it was reformatted by the plugin (sorry now I checked and see there is a setting how to turn it off)

**Comment by irengrig** [comment]
Yep, actually it was reformatted by the plugin (sorry now I checked and see there is a setting how to turn it off)

**Comment by dslomov** [comment]
Let's undo this reformatting in this PR and then reformat the entire repo with buildifier (https://github.com/bazelbuild/buildtools/)

**Comment by irengrig** [comment]
Done, sorry again

**Pull request merged**